### PR TITLE
feat: moderinze setup to use labels

### PR DESCRIPTION
**Pull request by esolitos**
## what

Modernizes the module setup
- Added the use of `cloudposse/label/null` for standardized naming
- Ran `make` actions to sync, as [recommended in another PR](https://github.com/cloudposse/terraform-aws-ecs-cluster/pull/11).
- Updated examples/complete to a better standard

⚠️ Changes are not backward compatible, resources will be recreated and their name will change.

## why

- elevate the module to the same standard as the other cloudposse modules 😃 
 
## references

closes: #3

### CODE REVIEW

File: alarms.tf [hunk 1]
```diff
@@ -4,13 +4,24 @@ locals {
     PercentIOLimitThreshold     = "${min(max(var.percent_io_limit_threshold, 0), 100)}"
   }
 
-  alert_for     = "efs"
-  sns_topic_arn = var.sns_topic_arn == "" ? aws_sns_topic.default.arn : var.sns_topic_arn
-  endpoints     = distinct(compact(concat(list(local.sns_topic_arn), var.additional_endpoint_arns)))
```
Comment (Gowiem) [inline alarms.tf:9]: You're removing `endpoints` and then using it below, won't this blow up?

Comment (Gowiem) [reply alarms.tf:9]: Ah you're defining in `main` -- Okay, all good to ignore this comment.

File: main.tf [hunk 2]
```diff
@@ -1,21 +1,40 @@
-data "aws_caller_identity" "default" {}
+locals {
+  e = module.this.enabled # Shorthand for quick reference
```
Comment (Gowiem) [inline main.tf:2]: We call this `enabled` in all cases. Let's update that 👍  

```suggestion
  enabled = module.this.enabled
```

File: main.tf [hunk 3]
```diff
@@ -1,21 +1,40 @@
...
+
+  sns_topic_policy_enabled = local.e && !var.add_sns_policy && var.sns_topic_arn != ""
+  sns_topic_arn            = var.add_sns_policy && var.sns_topic_arn != "" ? var.sns_topic_arn : join("", aws_sns_topic.default.*.arn)
+  endpoints                = distinct(compact(concat([local.sns_topic_arn], var.additional_endpoint_arns)))
+}
+
+data "aws_caller_identity" "default" {
+  count = module.this.enabled ? 1 : 0
```
Comment (Gowiem) [inline main.tf:10]: If we're going to define the local, let's change all usages of `module.this.enabled` to use `local.enabled` instead.

**Review by Gowiem — CHANGES_REQUESTED** [review]
@esolitos couple requests, but generally this is looking good. This will need an elevated approval, but I'll give my approval once changes are made and then ask the CP team to give it a review as well. Thanks!

File: main.tf (hunk 2 above)
Comment (esolitos) [reply main.tf]: 👍🏼  Updated

File: main.tf (hunk 3 above)
Comment (esolitos) [reply main.tf]: Good suggestion, I also added the usage of `count = local.enabled ? 1 : 0` in the resources which were missing it.

**Comment by esolitos** [comment]
I believe I fixed the failures reported by the CI. :)

**Comment by esolitos** [comment]
🙃  Or not.
I did the mistake of running the `make` script as first step, let me fix that.

**Comment by esolitos** [comment]
Not sure what is failing.

Is it because the example is in [./examples/simple](https://github.com/cloudposse/terraform-aws-efs-cloudwatch-sns-alarms/tree/main/examples/simple) instead of `./examples/complete`?

If yes, is renaming this in scope for this PR? 😅  I have no problem doing it, takes me 1 minute, I'm jsut wondering.

**Comment by Gowiem** [comment]
@esolitos yeah it looks like this module is so old it wasn't following where we put our primary example path (which is ancient) and therefore our newer CI pipes are failing as they expect that path. I'd move simple to complete and let's give it another run 👍

**Comment by Gowiem** [comment]
Ah I think we need a `versions.tf` file for `examples/complete` like we have here: https://github.com/cloudposse/terraform-aws-transit-gateway/blob/main/examples/complete/versions.tf

It may be that we need more than that too... We'll see what the test harness does. This is an old module that hasn't seen love in a long while... so you picked a good one 😁

**Comment by esolitos** [comment]
I did update the examples/complete with some additional love'n'care. 😄

**Comment by Gowiem** [comment]
Haha I didn't realize there isn't even a `test/src` directory... @esolitos if we want this to pass our standard tests we would need to go all the way towards adding Terratest to this repo like in https://github.com/cloudposse/terraform-aws-transit-gateway/tree/main/test OR if I should just pull in Cloud Posse folks and see if they can get around to reviving this module from what you've already done.

**Comment by Gowiem** [comment]
We still don't have terratests on this repo, so no use in trying to run them. I may take a crack at adding them at some point this week or next if I have the time. Thanks for the patience @esolitos!

**Pull request closed without merging**